### PR TITLE
FRO 168 download csv

### PR DESCRIPTION
**Pull request by PatFawbertMills**
- new hook to contain download logic and state
- button on search page gives feedback for: loading, success and error
- csv is downloaded

### CODE REVIEW

File: src/api/http-common.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import axios, { AxiosInstance, AxiosResponse } from "axios";
 
+interface ValidationError {
```
Comment (diversemix) [inline src/api/http-common.ts:3]: Am I being McThickThick? Where is this used?

File: src/hooks/useDownloadCsv.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  headers: {
+    accept: string;
+    "Content-Type": string;
+  };
+};
+
+const downloadFile = ({ data, fileName, fileType }) => {
+  const blob = new Blob([data], { type: fileType });
+
+  const a = document.createElement("a");
```
Comment (diversemix) [inline src/hooks/useDownloadCsv.ts:16]: Do we need to make sure this is hidden? Or if we do is the link not clickable? Have we tried this by enabling a mock of a slow connection in the browser debug? (to see how it behaves)

File: src/hooks/useDownloadCsv.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  return true;
+}
+
+export function useDownloadCsv() {
+  const [status, setStatus] = useState<TLoadingStatus>("idle");
+
+  const download = async (query: TRouterQuery) => {
+    setStatus("loading");
+    const download = await getDownloadCsv(query);
```
Comment (diversemix) [inline src/hooks/useDownloadCsv.ts:60]: Just to point out we have the var `download` hidden at this point (line 58)

**Review by diversemix — APPROVED** [review]
If it works - great ... some questions which I'm happy you resolve if you're good :smile:

File: src/api/http-common.ts (hunk 1 above)
Comment (PatFawbertMills) [reply src/api/http-common.ts:3]: You certainly aren't - it is a legacy of me attempting to cast the axios error which I learnt is not a good idea

File: src/hooks/useDownloadCsv.ts (hunk 3 above)
Comment (PatFawbertMills) [reply src/hooks/useDownloadCsv.ts]: Good spot

**Review by diversemix — APPROVED** [review]
Thanks for the education :brain:

**Pull request merged**